### PR TITLE
Added candidate naming convention to the style guide

### PR DESCRIPTION
**Pull request by vyadh**
As per Discord, raised to help document the commonly used naming and make it easier for newbies to see normal usage and help staying consistent with standard library usage. Hopefully all standard practice and nothing controversial here. The foundations were taken from [this comment in Discord](https://discord.com/channels/601130461678272522/615329862395101194/941324199748853760)

This does feel a little verbose but I couldn't think of reducing it without removing the examples.

There was also a question about restructuring this page. Not sure I'm best placed to do that, but happy to give that a shot with more guidance.

### CODE REVIEW

**Review by fdncred — APPROVED** [review]
Some of this stuff is subtle. Nice work!

File: book/style_guide.md [hunk 1]
```diff
@@ -180,6 +180,141 @@ let selectedProfile = (
...
+let userId = 123
+
+def fetch-user [user-id: int] {
+  # ...
+}
+```
+
+#### Environment Variables
+
+**It's recommended to** use SCREAMING_SNAKE_CASE for environment variable names.
```
Comment (NotTheDr01ds) [inline book/style_guide.md:300]:
```suggestion
With the exception of the pre-defined variables in `$env.config`, **it's recommended to** use SCREAMING_SNAKE_CASE for environment variable names.
```

File: book/style_guide.md [hunk 2]
```diff
@@ -180,6 +180,141 @@ let selectedProfile = (
...
+Incorrect:
+
+```nu
+fetch_user --id 123
+fetchUser --id 123
+```
+
+See also [Naming Commands](custom_commands.md#naming-commands).
+
+#### Sub-Commands
```
Comment (NotTheDr01ds) [inline book/style_guide.md:227]: I *think* we could combine the "Commands" and "Subcommands" sections and say something like:

> **It's recommended to** use kebab-case for command and subcommand names with multiple words.
>
> Subcommands are commands that are logically grouped under a parent command and separated from the command name by a space.

File: book/style_guide.md [hunk 3]
```diff
@@ -180,6 +180,141 @@ let selectedProfile = (
...
+
+Sub commands are commands that are logically grouped under a parent command and separated by a space.
```
Comment (NotTheDr01ds) [inline book/style_guide.md:229]:
```suggestion
Subcommands are commands that are logically grouped under a parent command and separated by a space.
```

**Comment by NotTheDr01ds** [comment]
Looks really good - A few comments, but nothing major.

I'd be fine with it "as is" and then I could go in and do some word-smithing (not necessarily on your stuff, but there are some general things that need work).

One question - How about "record key/table column names"?  I think it should be something like:

> Recommended:  snake_case, except when the name is meant for display/output.

In other words, a table column might be *"Totals"* if that was the intended *output* format.

But again, it's something I can add later - I just kind of want to if folks have any opinions on that topic as well.

**Comment by NotTheDr01ds** [comment]
@fdncred If you'd like, you can merge this as it stands today. I don't think there are any substantial issues with it that should hold it back.  I can then go in and do touch-up on it after that?

**Review by sholderbach — APPROVED** [review]
No lies detected :) Let's merge

**Pull request merged**